.github/workflows/deploy.yml

```diff
@@ -229,7 +230,7 @@ jobs:
   finishing-prep:
     name: Finishing Prep
     needs: [deploy-infra, begin-deployment, deploy-app]
-    if: always()
+    if: always() && needs.deploy-infra.result == 'success' && needs.deploy-app.result == 'success'
     runs-on: ubuntu-latest
```

Comment:
This isn't related to the feature, but I was running into issues where `finishing-prep` was running before other things and causing issues. I even saw the container hang for a good 30 minutes for a reason I'm still unsure of.

Reply:
It was also happening with Cypress, so I added some conditions there as well.